fbw-common/src/wasm/systems/systems/src/hydraulic/flap_slat.rs

```diff
@@ -121,17 +123,97 @@ impl Actuator for FlapSlatHydraulicMotor {
     }
 }
 
-pub struct FlapSlatAssembly {
-    position_left_percent_id: VariableIdentifier,
-    position_right_percent_id: VariableIdentifier,
+pub struct SecondarySurface {
+    surface_position_ids: Vec<VariableIdentifier>,
+    surface_angle_ids: Vec<VariableIdentifier>,
 
-    angle_left_id: VariableIdentifier,
-    angle_right_id: VariableIdentifier,
+    surface_position_overall_id: VariableIdentifier,
+    surface_angle_overall_id: VariableIdentifier,
 
+    surface_positions: Vec<Ratio>,
+    surface_angles: Vec<Angle>,
+}
+impl SecondarySurface {
+    // side is LEFT|RIGHT
+    // surface is SLATS|FLAPS
+    pub fn new(
+        context: &mut InitContext,
+        side: &str,
+        surface: &str,
+        number_of_surfaces: usize,
+    ) -> Self {
+        let mut surface_position_ids = Vec::with_capacity(number_of_surfaces);
+        let mut surface_angle_ids = Vec::with_capacity(number_of_surfaces);
+        let mut surface_positions = Vec::with_capacity(number_of_surfaces);
+        let mut surface_angles = Vec::with_capacity(number_of_surfaces);
+
+        for p in 1..number_of_surfaces + 1 {
+            let surface_position_id =
+                context.get_identifier(format!("{}_{}_{}_POSITION_PERCENT", side, surface, p));
+            surface_position_ids.push(surface_position_id);
+
+            let surface_angle_id =
+                context.get_identifier(format!("{}_{}_{}_ANGLE", side, surface, p));
+            surface_angle_ids.push(surface_angle_id);
+
+            surface_positions.push(Ratio::default());
+            surface_angles.push(Angle::default());
+        }
+
+        let surface_position_overall_id =
+            context.get_identifier(format!("{}_{}_POSITION_PERCENT", side, surface));
+        let surface_angle_overall_id =
+            context.get_identifier(format!("{}_{}_ANGLE", side, surface));
+
+        Self {
+            surface_position_ids,
+            surface_angle_ids,
+            surface_positions,
+            surface_angles,
+            surface_position_overall_id,
+            surface_angle_overall_id,
+        }
+    }
+
+    // No failures are considered. All surfaces have the same deflection.
+    fn update(&mut self, position: &Ratio, angle: &Angle) {
+        for idx in 0..self.surface_positions.len() {
+            self.surface_positions[idx] = *position;
+            self.surface_angles[idx] = *angle;
+        }
+    }
+}
+impl SimulationElement for SecondarySurface {
+    fn write(&self, writer: &mut SimulatorWriter) {
+        for idx in 0..self.surface_positions.len() {
+            let id = self.surface_position_ids[idx];
+            let position = self.surface_positions[idx];
+            writer.write(&id, position.get::<percent>());
+        }
```

Comment:
This is much more idiomatic Rust:
```suggestion
        for (id, position) in self
            .surface_position_ids
            .iter()
            .zip(&self.surface_positions)
        {
            writer.write(id, position.get::<percent>());
        }
```